### DISCUSSION TITLE
Variant with theme property in component with styled context

### DISCUSSION THREAD

**Discussion by Maciej-Siara** [tamagui/tamagui#1578] [General]
### Current Behavior

Variant with theme property in styled context component does not change component color palette.
Without styled context theme property changes colors in the correct way

### Expected Behavior

Variant with theme property in styled context component should change component color palette in the same way as component without styled context.

### Tamagui Version

```markdown
1.55.2
```

### Reproduction

```markdown
https://codesandbox.io/p/sandbox/green-flower-rtqkl4?file=/packages/app/features/home/screen.tsx:35,14
```

### System Info

```markdown
System:
    OS: macOS 13.3
    CPU: (8) arm64 Apple M2
    Memory: 62.80 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    Yarn: 3.5.0 - /usr/local/bin/yarn
    npm: 9.6.6 - ~/.nvm/versions/node/v18.15.0/bin/npm
  Browsers:
    Chrome: 116.0.5845.96
    Firefox: 112.0.1
    Safari: 16.4
  npmPackages:
    @babel/runtime: ^7.18.9 => 7.21.0 
    @manypkg/cli: ^0.19.1 => 0.19.2 
    check-dependency-version-consistency: ^3.0.3 => 3.0.3 
    eslint: ^8.21.0 => 8.39.0 
    node-gyp: ^9.3.1 => 9.3.1 
    prettier: ^2.7.1 => 2.8.8 
    turbo: ^1.10.3 => 1.10.3 
    typescript: ^5.1.3 => 5.1.3
```

**Comment by natew** [comment]
We don't support theme in variants, just for complexity sake at the moment. Converting to discussion as we'd need to see widespread support for this before implementing, it has a cost to runtime size and complexity.
